gunrock/app/knn/knn_helpers.cuh

```diff
@@ -24,9 +24,53 @@ namespace knn {
 
 template <typename ValueT>
 __device__ __host__ ValueT _sqrt(const ValueT &a) {
-    return (ValueT) sqrtf((float)a);
+    return sqrt(a);
 }
```

Comment:
We may have to support both float and double for this:

```c++
template <typename ValueT>
__device__ __host__ ValueT _sqrt(const float &a) {
    return sqrtf((float)a);
}

template <typename ValueT>
__device__ __host__ ValueT _sqrt(const double &a) {
    return sqrt((double)a);
}
```

Reply:
This does not work

Reply:
I am sorry, why does this not work?

Reply:
How shall I deal with the case when I have:
```
ValueT x = fun();
  _sqrt(x);
```
? The above function does not work on it (compiler shows up error). Why I cannot leave how is right now?

Reply:
I thought `sqrt` was for `double` only, so you'd feed to cast it to a double then if you want to use it for any `ValueT`.

Reply:
If it is for double only then I do not need the first function you posted above, right? Do you want me to do only for doubles and then cast each value to double before using `_sqrt`?